[Qt] Update to 6.9.2

kimbirkelund opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

This is my first PR updating a port. I've modeled it on [this PR updating Qt to 6.9.1](https://github.com/microsoft/vcpkg/pull/46892).

kimbirkelund committed:
Bump vcpkg.jsons.

kimbirkelund committed:
Bump version.

kimbirkelund committed:
Update hashes.

kimbirkelund committed:
Add versions.

petersteneteg commented:
Looking forward to this update, since qtbase 6.9.1 fails to build with Xcode 26!

kimbirkelund committed:
Adjust ffmpeg-compile-def.patch.

kimbirkelund committed:
Delete patch as it has been applied in repo.

kimbirkelund committed:
Remove non-matching parts of static_find_modules.patch.

Removes a great chunk that seems to have reworked in the repo to no
longer be applicable. But someone with more knowledge of the
qtmultimedia port should look at this.

kimbirkelund committed:
Adjust static_find_modules.patch.

kimbirkelund committed:
Delete patch as the applicable code seems to no longer exist.

kimbirkelund committed:
Adjust 0001-devendor-meshoptimizer.patch.

kimbirkelund committed:
Adjust ffmpeg.patch.

kimbirkelund committed:
HAVE_Assimp is reported unused and does not occur in qtquick3d repo.

kimbirkelund committed:
Remove unnecessary patch.

kimbirkelund committed:
Update versions.

kimbirkelund committed:
Add patch to explicitly convert DWORD to qsizetype.

kimbirkelund committed:
Update versions.

kimbirkelund commented:
The `x86-windows`, `arm64-osx` and `x64-osx` build is failing on `qtbase[sql-oci]` with the following error:

```
CMake Error at cmake/QtBuildInformation.cmake:554 (message):
  Feature "sql_oci": Forcing to "ON" breaks its condition:

      Oracle_FOUND

  Condition values dump:

      Oracle_FOUND = "FALSE"

Call Stack (most recent call first):
  cmake/QtFeature.cmake:786 (qt_configure_add_report_error)
  cmake/QtFeature.cmake:923 (qt_feature_check_and_save_internal_value)
  cmake/QtFeature.cmake:1192 (qt_evaluate_feature)
  src/plugins/sqldrivers/CMakeLists.txt:46 (qt_feature_module_end)
```

I don't know what to do about that.

dg0yt commented:
The unexpected error is `qtinterfaceframework`. 
~~~
-- Fetching https://code.qt.io/qt/qtinterfaceframework.git d7f7b3e722070d056dff0ea098bda3d098dc991a...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /usr/bin/git fetch https://code.qt.io/qt/qtinterfaceframework.git d7f7b3e722070d056dff0ea098bda3d098dc991a --depth 1 -n
    Working Directory: /vcpkg/downloads/git-tmp
    Error code: 128
    See logs for more information:
      /mnt/vcpkg-ci/b/qtinterfaceframework/git-fetch-x64-linux-err.log
~~~
Raw git? Flaky by design. No asset caching.

kimbirkelund commented:
Thanks for the input. Is there an easy way to see which should just be ignored? Since both the "real" and ignored errors show up in the log artifacts?

Also `qt3d[vulkan]` fails on mac, because it depends on `qtbase[vulkan]` which is expected to fail. Does that mean that `qt3d[vulkan]` is implicitly ignored, or should that be added to the baseline file?

dg0yt commented:
> Is there an easy way to see which should just be ignored? Since both the "real" and ignored errors show up in the log artifacts?

In the AZP output for x64-windows-static-md, I cannot see explict hints for expected failures, but there is

~~~
There are some feature test problems!
D:\a\_work\1\s\scripts\azure-pipelines/../ci.feature.baseline.txt: error: qtinterfaceframework[core]:x64-windows-static build failed but was expected to pass
note: consider adding `qtinterfaceframework=fail`, or `qtinterfaceframework:x64-windows-static=fail`, or equivalent skips
note: if some features are required, consider effectively always enabling those parts in portfile.cmake for qtinterfaceframework, or consider adding `qtinterfaceframework[required-feature]=options` to include 'required-feature' in all tests
~~~

The expectedd failures are in `scripts/ci.*`.

adriweb commented:
Note that 6.9.3 fixes building on macOS 26 due to a framework being linked that became unavailable.
Might be worth looking into updating to this version directly.

xavier2k6 commented:
Qt 6.10.0 is officially out now!
